### PR TITLE
React part 2

### PR DESCRIPTION
**Pull request by speudusa**

### CODE REVIEW

File: content/modern-js/reading/es6-syntax/index.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ lastEditorGitHub: # update any time edits are made after review
 lastMod: # UPDATE ANY TIME CHANGES ARE MADE
 ---
```
Comment (codinglikeagirl42) [inline content/modern-js/reading/es6-syntax/index.md:14]: Please update the MetaData. 
lastEditor:
lastEditorGitHub:
lastMod:

File: content/react-part-2/reading/conditional-rendering/index.md [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+weight: 2
+originalAuthor: Courtney Frey # to be set by page creator
+originalAuthorGitHub: speudusa # to be set by page creator
+reviewer: Kimberly Horan # to be set by the page reviewer
+reviewerGitHub: codinglikeagirl42 # to be set by the page reviewer
+lastEditor: # update any time edits are made after review
+lastEditorGitHub: # update any time edits are made after review
+lastMod: # UPDATE ANY TIME CHANGES ARE MADE
+---
+TODO: link to conditionals chapter in JS
```
Comment (codinglikeagirl42) [inline content/react-part-2/reading/conditional-rendering/index.md:14]: Suggestion -> make this a comment, so you can still find it but it doesn't appear as a line in the book.

File: content/react-part-2/reading/js-in-react/index.md [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+
+{{% expand "Check the output" %}}
+
+![ShinyButton Output](./images/ShinyButtonOutput.png)
+
+{{% /expand %}}
+
+
+### Working with Props and JS
+//TODO: format as relref link
```
Comment (codinglikeagirl42) [inline content/react-part-2/reading/js-in-react/index.md:158]: Suggestion - remove once link is working below or make it a comment.

File: content/react-part-2/reading/js-in-react/index.md [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+
+Recall from the [PREVIOUS CHAPTER](react-lsn1/reading/components/#functional-components), `props` are arguments passed to and between functions.  We can use anything as a `props`, since `props` are placeholders.
```
Comment (codinglikeagirl42) [inline content/react-part-2/reading/js-in-react/index.md:160]: Suggestion not to capitalize PREVIOUS CHAPTER just use previous chapter.  And please fix the link it is not working.

File: content/react-part-2/reading/conditional-rendering/index.md [hunk 5]
```diff
@@ -0,0 +1,452 @@
...
+
+We can use `if` statements to create conditions.  Those conditions then determine the output of our application.
+
+In the example below, we have two functions, `Rainy` and `Sunny`.  Each one returns a simple `<h1>` message.
+
+In the `App` function, we use an `if` statement and `props` to render which statement will appear in the browser.  The `if` block starts at line 15, and declares `isRainy` as `true`.  This will return the message from the `Rainy` function.  When `isRainy` is not `true`, it will return the ` <h1>No rain today!</h1>` message instead.
+
+{{% notice blue "Example Setting our Conditional to True" "rocket" %}}
+
+TODO: I've been using codesandbox for testing my examples -- would this be an option for students or too distracting?  the react.dev documentation uses codesandbox
```
Comment (codinglikeagirl42) [inline content/react-part-2/reading/conditional-rendering/index.md:28]: Suggestion -> make this a comment, so you can still find it but it doesn't appear as a line in the book.

File: content/react-part-2/reading/conditional-rendering/index.md [hunk 6]
```diff
@@ -0,0 +1,452 @@
...
+         <RainOrShine 
+            isRainy={false}
+         />
+      </div>
+      );
+   }
+   ```
+{{% /notice %}}
+
+//TODO: Also a codesandbox feature.  Click the black left border.  Then click the hamburger menu in the top left corner.  Select `App.js` to explore.
```
Comment (codinglikeagirl42) [inline content/react-part-2/reading/conditional-rendering/index.md:114]: Suggestion -> make a comment so it doesn't print to page or:

```suggestion
Click the black left border.  Then click the hamburger menu in the top left corner.  Select `App.js` to explore.
```

File: content/react-part-2/reading/conditional-rendering/index.md [hunk 7]
```diff
@@ -0,0 +1,452 @@
...
+
+{{% expand "Output" %}}
+   ## Bring your umbrella!
+{{% /expand %}}
+
+
+## Other applications for Conditional Rendering
+
+Conditional Rendering can also be applied to the styling of elements. 
+Let's add some to our `RainOrShine` function.  We will need
```
Comment (codinglikeagirl42) [inline content/react-part-2/reading/conditional-rendering/index.md:289]: Question ?  We will need What??  Found this confusing.

File: content/react-part-2/reading/mapping/index.md [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+
+We can use `map()` for more complicated arrays, such as an array of objects. Like the `for` loop, you can nest the `map()` function.
+
+<!-- //TODO: link to objects page in JS textbook will need to update when JS added to this book-->
+
+With all of the flexibility of the `map()` function, we need to ensure it renders the correct element. 
+
+The key becomes a value's unique id that React associates with its component.  This helps prevent values being mixed up between components.  This unique id is crucial if the items in your array can move, be inserted into other components, or even deleted.  
+
+It is not advised to create keys while the application is running.  You should include them in your data somehow. A key needs to be a string or number datatype.  If you have the opportunity to create your data, include some sort of `id` key.  If your data does not have an `id` then use what is available.
```
Comment (codinglikeagirl42) [inline content/react-part-2/reading/mapping/index.md:85]:
```suggestion
It is not advised to create keys while the application is running.  You should include them in your data somehow. A key needs to be a string or number data type.  If you have the opportunity to create your data, include some sort of `id` key.  If your data does not have an `id` then use what is available.
```

File: content/react-part-2/studio/index.md [hunk 9]
```diff
@@ -0,0 +1,243 @@
...
+The app you complete in this studio will have the following functionality:
+   1. A star rating component that can be updated to display 0 - 5 stars.
+   1. The ability to mimic the Save vs Saved button on a Pinterest pin using conditional rendering.
+   1. Displaying the recipe information using the `map` function and a JSON file.
+   1. CSS styling has been created for this project. The instructions will let you know what and how to style elements.  
+
+
+## Getting Started
+
+1. Fork and clone the [starter code ADD LINK](). //TODO: Link to starter code
```
Comment (codinglikeagirl42) [inline content/react-part-2/studio/index.md:27]: Add Link.

**Review by codinglikeagirl42 — APPROVED** [review]
Looks good, a few suggestions to look at then merge away.

Also of note- codesandbox - seems spelling according to their site is CodeSandbox.

File: content/react-part-2/reading/conditional-rendering/index.md (hunk 2 above)
Comment (speudusa) [reply content/react-part-2/reading/conditional-rendering/index.md:14]: i think i will pull this for now. if we need a link back, we can add it once the rest of the JS is in the book.

File: content/react-part-2/reading/conditional-rendering/index.md (hunk 7 above)
Comment (speudusa) [reply content/react-part-2/reading/conditional-rendering/index.md:292]: well, that was a cliffhanger for sure!  added more information to make this less suspensful.

Comment (speudusa) [reply content/react-part-2/reading/conditional-rendering/index.md:292]: (line 298 - not sure if this comment is lined up correctly right now)

File: content/react-part-2/reading/js-in-react/index.md (hunk 4 above)
Comment (speudusa) [reply content/react-part-2/reading/js-in-react/index.md:160]: link fixed! woot!

File: content/react-part-2/studio/index.md (hunk 9 above)
Comment (speudusa) [reply content/react-part-2/studio/index.md]: check check!

**Pull request merged**